src/parallel/DomainDecompMPIBase.cpp

```diff
@@ -211,12 +211,12 @@ void DomainDecompMPIBase::assertDisjunctivity(ParticleContainer* moleculeContain
 		unsigned long num_molecules = moleculeContainer->getNumberOfParticles();
 		std::vector<unsigned long> tids(num_molecules);
 
-		int i = 0;
+		unsigned long i = 0;
 		for (auto m = moleculeContainer->iterator(ParticleIterator::ONLY_INNER_AND_BOUNDARY); m.isValid(); ++m) {
 			tids[i] = m->getID();
 			i++;
 		}
-		MPI_CHECK(MPI_Send(tids.data(), num_molecules, MPI_UNSIGNED_LONG, 0, 2674 + _rank, _comm));
+		MPI_CHECK(MPI_Send(tids.data(), i, MPI_UNSIGNED_LONG, 0, 2674 + _rank, _comm));
```

Comment:
That looks suspicious. Shouldn't 'i' and `num_molecules` be the same here? And if they are not we need at least a comment why that is the case (and probably that we have a bug somewhere?)

Reply:
So yeah, I guess I found the reason. Printing the relevant values at line 218 gives the following output:

INFO:	20220222T131412 0.031569 [3]	i = 5058 num_molecules = 9438 tids.size() = 9438
INFO:	20220222T131412 0.031676 [1]	i = 5355 num_molecules = 9438 tids.size() = 9438
INFO:	20220222T131412 0.031965 [2]	i = 5355 num_molecules = 9438 tids.size() = 9438

If we replace line 211 with unsigned long num_molecules = moleculeContainer->getNumberOfParticles(ParticleIterator::ONLY_INNER_AND_BOUNDARY); , that fixes the bug.

It is odd that this doesn't seem to trigger on other machines.

Reply:
you are only looking at the AutoPas version, right? That is not (yet) tested enough. I just discovered that there seem to be some weird things going on in `AutoPasContainer::getNumberOfParticles()`. So maybe check if you can fix things there.

Reply:
> If we replace line 211 with unsigned long num_molecules = moleculeContainer->getNumberOfParticles(ParticleIterator::ONLY_INNER_AND_BOUNDARY); , that fixes the bug.

Makes sense btw because if no argument is given we default to `ALL_CELLS`